delta/plugins/storage/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/storages/storage/StoragesSpec.scala

```diff
@@ -0,0 +1,661 @@
+        val current = currentState(dId, project, diskVal, Json.obj(), tags = Map(tag1 -> 2), rev = 2)
+
+        next(Initial, event) shouldEqual Initial
+
+        next(current, event) shouldEqual
+          current.copy(rev = 3, updatedAt = time2, updatedBy = alice, tags = Map(tag1 -> 2, tag2 -> 1))
+      }
+
+      "from a new StorageDeprecated event" in {
+        val event   = StorageDeprecated(dId, project, 2, time2, alice)
+        val current = currentState(dId, project, diskVal, Json.obj())
+
+        next(Initial, event) shouldEqual Initial
+
+        next(current, event) shouldEqual current.copy(rev = 2, deprecated = true, updatedAt = time2, updatedBy = alice)
+      }
+    }
+  }
+
+  "The Storages operations bundle" when {
+    val uuid                                  = UUID.randomUUID()
+    implicit val uuidF: UUIDF                 = UUIDF.fixed(uuid)
+    implicit val rcr: RemoteContextResolution = RemoteContextResolution.fixed(
+      Vocabulary.contexts.metadata -> jsonContentOf("/contexts/metadata.json"),
+      contexts.storage             -> jsonContentOf("/contexts/storages.json")
+    )
+    implicit val caller: Caller               = Caller(bob, Set(bob, Group("mygroup", realm), Authenticated(realm)))
+    implicit val baseUri: BaseUri             = BaseUri("http://localhost", Label.unsafe("v1"))
+
+    val org                      = Label.unsafe("org")
+    val orgDeprecated            = Label.unsafe("org-deprecated")
+    val base                     = nxv.base
+    val project                  = ProjectGen.project("org", "proj", base = base, mappings = ApiMappings.default)
+    val deprecatedProject        = ProjectGen.project("org", "proj-deprecated")
+    val projectWithDeprecatedOrg = ProjectGen.project("org-deprecated", "other-proj")
+    val projectRef               = project.ref
+    val storageConfig            = StoragesConfig(aggregate, keyValueStore, pagination, indexing, config)
+
+    val tag           = Label.unsafe("tag")
+    val diskPayload   = jsonContentOf("storage/disk-storage.json")
+    val s3Payload     = jsonContentOf("storage/s3-storage.json")
+    val remotePayload = jsonContentOf("storage/remote-storage.json")
+
+    def projectSetup =
+      ProjectSetup
+        .init(
+          orgsToCreate = org :: orgDeprecated :: Nil,
+          projectsToCreate = project :: deprecatedProject :: projectWithDeprecatedOrg :: Nil,
+          projectsToDeprecate = deprecatedProject.ref :: Nil,
+          organizationsToDeprecate = orgDeprecated :: Nil
+        )
+
+    val storages = (for {
+      eventLog         <- EventLog.postgresEventLog[Envelope[StorageEvent]](EventLogUtils.toEnvelope).hideErrors
+      (orgs, projects) <- projectSetup
+      storages         <- Storages(storageConfig, eventLog, perms, orgs, projects, access)
+    } yield storages).accepted
+
+    "creating a storage" should {
+
+      "succeed with the id present on the payload" in {
```

Comment:
maybe a test to for changing the default?

Reply:
This is already on the tests. The SSEs test verify that, the test on the update verifies that (because the revision gets affected), and others